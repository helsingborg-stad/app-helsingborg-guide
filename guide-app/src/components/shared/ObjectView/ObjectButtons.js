import React, { memo, useState } from "react";
import { Text, View } from "react-native";
import { useDispatch } from "react-redux";
import Icon from "react-native-vector-icons/Entypo";
import { selectCurrentContentObject } from "@actions/uiStateActions";
import { Colors } from "@assets/styles";
import { trackScreen } from "@utils/MatomoUtils";
import styles from "./style";

type Props = {
  swipeable?: Boolean,
  scrollable?: Boolean,
  panToIndex?: number,
  navigation: Object,
  array?: Array,
  order?: number,
  path?: string,
};
const ObjectButtons = (props: Props) => {
  const { array, navigation, order, scrollable, panToIndex, path } = props;
  const [width, setWidth] = useState("");
  const split = path.split("/");
  const dispatch = useDispatch();

  const onLayout = (event) => {
    setWidth(event.nativeEvent.layout.width);
  };

  return array?.length ? (
    <View style={styles.navGuideWrapper}>
      <Text style={styles.navGuideBarStep}>{`${order + 1} av ${
        array.length
      }`}</Text>
      <View style={styles.navGuide}>
        <Icon
          name={"chevron-left"}
          size={36}
          color={Colors.themeExtra1}
          style={{ opacity: order > 0 ? 1 : 0.4 }}
          onPress={
            order > 0
              ? () => {
                  let newPath =
                    split.slice(0, split.length - 1).join("/") +
                    `/${array[order - 1].title}`;
                  trackScreen(newPath, newPath);
                  scrollable && scrollable(order - 1);
                  dispatch(selectCurrentContentObject(array[order - 1]));
                  panToIndex && panToIndex(order - 1);
                  navigation.navigate({
                    name: "ObjectScreen",
                    params: {
                      title: array[order - 1].title,
                      currentGuide: array[order - 1],
                      order: order - 1,
                      array: array,
                      swipeable: true,
                      path: newPath,
                    },
                    merge: true,
                  });
                }
              : null
          }
        />
        <View style={styles.navGuideBarWrapper}>
          <View onLayout={onLayout} style={styles.navGuideBar}>
            {width ? (
              <View
                style={[
                  styles.navGuideBarFilled,
                  {
                    transform: [
                      {
                        translateX:
                          -width +
                          Math.round(width * ((order + 1) / array.length)),
                      },
                    ],
                  },
                ]}
              />
            ) : null}
          </View>
        </View>
        <Icon
          name={"chevron-right"}
          size={36}
          color={Colors.themeExtra1}
          style={{ opacity: order + 1 !== array.length ? 1 : 0.5 }}
          onPress={
            order + 1 !== array.length
              ? () => {
                  let newPath =
                    split.slice(0, split.length - 1).join("/") +
                    `/${array[order + 1].title}`;
                  trackScreen(newPath, newPath);
                  scrollable && scrollable(order + 1);
                  dispatch(selectCurrentContentObject(array[order + 1]));
                  panToIndex && panToIndex(order + 1);
                  navigation.navigate({
                    name: "ObjectScreen",
                    params: {
                      title: array[order + 1].title,
                      currentGuide: array[order + 1],
                      order: order + 1,
                      array: array,
                      swipeable: true,
                      path: newPath,
                    },
                    merge: true,
                  });
                }
              : null
          }
        />
      </View>
    </View>
  ) : null;
};

export default memo(ObjectButtons);
