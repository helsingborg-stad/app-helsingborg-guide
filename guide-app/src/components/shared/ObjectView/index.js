// @flow

import React, { Component, useEffect, useRef, useState, Animated} from "react";
import {SafeAreaView, View, Text, ScrollView, Linking } from "react-native";

import styles from "./style";
import { Colors } from "@assets/styles";
import SharingService from "@services/SharingService";
import ButtonsBar from "@shared-components/btn_bar";
import ButtonsBarItem from "@shared-components/btn_bar_item";
import ImageSwiper from "@shared-components/ImageSwiper";
import LangService from "@services/langService";
import LinkTouchable from "@shared-components/LinkTouchable";
import AudioPlayerView from "@shared-components/AudioPlayerView";
import { openLink } from "@hooks/useOpenLink";
import Icon from "react-native-vector-icons/Entypo";

type Props = {
  contentObject: ContentObject,
  guideId?: number,
  imageIndex: number,
  guideType?: GuideType,
  onSwiperIndexChanged: (newIndex: number) => void,
  audioButtonDisabled: boolean,
  videoButtonDisabled: boolean,
  onGoToImage: (image: Images) => void,
  onGoToLink: (url: string, title?: string) => void,
  loadAudioFile: () => void,
  onGoToVideo: (video?: MediaContent) => void,
};

function displayID(searchableID: string) {
  const idText = (
    <View style={styles.idContainer}>
      <Text style={styles.idText}>{`ID #${searchableID}`}</Text>
    </View>
  );
  return idText;
}

function displayTitle(
  title: string,
  searchableID: string,
  guideType: ?GuideType,
) {
  return (
    <View>
      <View style={styles.titleContainer}>
        <Text style={styles.title}>{title}</Text>
      </View>
      {guideType === "guide" ? displayID(searchableID) : null}
    </View>
  );
}

function displayText(description?: string) {
  return <Text style={styles.article}>{description}</Text>;
}

function displayLinks(
  links: Link[],
  onGoToLink: (url: string, title?: string) => void,
) {
  return links.map((item, index) => (
    <LinkTouchable
      key={item.url || index}
      title={item.title}
      onPress={() => {
        // openLink(item.url);
        Linking.openURL(item.url);
      }}
    />
  ));
}

function displayButtonsBar(
  audio?: MediaContent,
  video?: MediaContent,
  audioButtonDisabled: boolean,
  videoButtonDisabled: boolean,
  loadAudioFile: () => void,
  onGoToVideo: (video?: MediaContent) => void,
) {
  const audioBtnInvisible = !audio || !audio.url;
  const videoBtnInvisible = !video || !video.url;

  if (videoBtnInvisible && audioBtnInvisible) {
    return null;
  }

  const audioBarItem = audioBtnInvisible ? null : (
    <ButtonsBarItem
      disabled={audioButtonDisabled}
      onPress={() => {
        loadAudioFile();
      }}
      name="headphones"
      color={Colors.themeSecondary}
      size={18}
      text={LangService.strings.LISTEN}
      view="row"
    />
  );

  const videoBarItem = videoBtnInvisible ? null : (
    <ButtonsBarItem
      disabled={videoButtonDisabled}
      onPress={() => {
        onGoToVideo(video);
      }}
      name="play-box-outline"
      color={Colors.themeSecondary}
      size={18}
      text={LangService.strings.VIDEO}
      view="row"
    />
  );

  return (
    <ButtonsBar>
      {audioBarItem}
      {videoBarItem}
    </ButtonsBar>
  );
}

const guideButtons = (props) => {
  const { array, navigation, selectObject, order} = props;
  const [ width, setWidth ] = useState("");

  const onLayout=(event)=> {
    setWidth(event.nativeEvent.layout.width)
  }

  return array?.length ? <View style={styles.navGuideWrapper}>
    <Text style={styles.navGuideBarStep}>{`${(order + 1)} av ${array.length}`}</Text>
    <View style={styles.navGuide}>
    <Icon
      name={"chevron-left"}
      size={36}
      color={Colors.themeExtra1}
      style={{opacity: order > 0 ? 1 : 0.4}}
      onPress={order > 0 ? () => {
        selectObject && selectObject(array[order - 1]);
        navigation.navigate("ObjectScreen", {
          title: array[order - 1].title,
          currentGuide: array[order - 1],
          array: array,
          order: order - 1,
        })
      } : null}
    />
    <View style={styles.navGuideBarWrapper}>
      <View
        onLayout={onLayout}
        style={styles.navGuideBar}>
        {width ? <View style={[styles.navGuideBarFilled,
          {
            transform: [
              {translateX: -width + Math.round(width * ((order + 1) / array.length))}
            ] }]} />  : null }
      </View>
    </View>
    <Icon
      name={"chevron-right"}
      size={36}
      color={Colors.themeExtra1}
      style={{opacity: (order + 1) !== array.length ? 1 : 0.5}}
      onPress={(order + 1) !== array.length ? () => {
        selectObject && selectObject(array[order + 1]);
        navigation.navigate("ObjectScreen", {
          title: array[order + 1].title,
          currentGuide: array[order + 1],
          order: order + 1,
          array: array,
        })
      } : null}
    />
    </View>
  </View> : null
};

/*
 * Underlying sharingservice needs a reference to a Component instance
 */
const ObjectView = (props) => {

    const { guideId } = props;
    console.log("the props", props)
    return (
      <View style={styles.viewContainer}>
      <SafeAreaView style={{flex: 1}}>
        <ScrollView style={styles.container}>
          <View
            style={styles.imageContainer}>
            <ImageSwiper
              sessionId={guideId}
              images={props.contentObject.images}
              onSwiperIndexChanged={props.onSwiperIndexChanged}
              onGoToImage={props.onGoToImage}
            />
            {props.contentObject.images[props.imageIndex] && (
              <View style={styles.shareBtn}>
                <SharingService
                  title={props.contentObject.title}
                  image={props.contentObject.images[props.imageIndex]}
                  sender={this}
                  shareType="share_object"
                />
              </View>
            )}
          </View>

          <View style={styles.bodyContainer}>
            {displayTitle(
              props.contentObject.title,
              props.contentObject.searchableId,
              props.guideType,
            )}
            {displayButtonsBar(
              props.contentObject.audio,
              props.contentObject.video,
              props.audioButtonDisabled,
              props.videoButtonDisabled,
              props.loadAudioFile,
              props.onGoToVideo,
            )}
            <View style={styles.articleContainer}>
              {props.contentObject.description
                ? displayText(props.contentObject.description)
                : null}
              {props.contentObject.links
                ? displayLinks(
                  props.contentObject.links,
                  props.onGoToLink,
                )
                : null}
            </View>
          </View>
        </ScrollView>
        <AudioPlayerView />
      </SafeAreaView>
      <SafeAreaView>
        <View style={styles.navGuideContainer}>
          {guideButtons(props)}
        </View>
        </SafeAreaView>
      </View>
    );
}

export default ObjectView;
