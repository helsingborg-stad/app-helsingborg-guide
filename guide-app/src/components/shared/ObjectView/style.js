import { StyleSheet } from "react-native";
import { Colors, TextStyles } from "@assets/styles";
import { StyleSheetUtils } from "@utils";


export default StyleSheet.create({
  viewContainer: {
    flex: 1,
    justifyContent: "flex-end",
    alignItems: "stretch",
    backgroundColor: Colors.white,
    overflow: "visible"
  },

  container: {
    ...StyleSheet.absoluteFillObject,
    flexDirection: "column",
    flex: 1
  },
  imageContainer: {
    flex: 1,
    width: "100%",
    height: "auto",
    aspectRatio: 16 / 9
  },
  bodyContainer: {
    minHeight: "100%",
    flex: 1,
    alignItems: "stretch"
  },
  scrollView: {
    paddingBottom: 70,
    flex: 1
  },
  infoContainer: {
    flexDirection: "row",
    paddingVertical: 20,
  },

  titleWrapper: {
    flex: 2,
    justifyContent: "center",
  },

  titleContainer: {
    paddingHorizontal: 20,
  },

  title: StyleSheetUtils.flatten([
    TextStyles.defaultFontFamily,
    {
      fontSize: 20,
      fontWeight: "400",
      lineHeight: 25
    }
  ]),
  idContainer: {
    paddingHorizontal: 20,
  },
  idText: StyleSheetUtils.flatten([
    TextStyles.defaultFontFamily,
    {
      fontSize: 16,
      fontWeight: "500",
      lineHeight: 22,
      color: Colors.themeSecondary
    }
  ]),
  articleContainer: {
    flex: 4,
    display: "flex",
    paddingHorizontal: 20,
    paddingVertical: 10,
    backgroundColor: 'rgba(248, 248, 248, 1)',
    overflow: "visible",
  },
  article: {
    fontSize: 14,
    lineHeight: 20,
  },

  shareBtn: {
    position: "absolute",
    bottom: 25,
    right: 25,
    zIndex: 50
  },
  navGuideContainer: {
    justifyContent: "flex-end",
    zIndex: 100,
    backgroundColor: "white",
    marginTop: 20,
  },

  navGuideWrapper: {
    flexDirection: "column",
    alignItems: "center",
    width: "100%",

  },

  navGuide: {
    display: "flex",
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-around",
    width: "100%"
  },

  navGuideBarWrapper: {
    width: "60%",
    flexDirection: "column"
  },
  navGuideBarStep: {
    textAlign: "center"
  },

  navGuideBar: {
    height: 29,
    backgroundColor: "#EAEAEA",
    borderRadius: 20,
    overflow: "hidden"
  },

  navGuideBarFilled: {
    height: "100%",
    width: "100%",
    backgroundColor: "#711A5B",
    borderRadius: 20,
    overflow: "hidden"
  }
});
