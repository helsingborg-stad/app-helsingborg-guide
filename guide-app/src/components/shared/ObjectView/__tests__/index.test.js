// @flow

/* eslint-disable import/first */

import "react-native";
import React from "react";
import ObjectView from "../index";

import renderer from "react-test-renderer";

const currentContentObject: ContentObject = {
  id: "59521803842fa",
  images: [
    {
      large:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-01_bottenv%C3%A5ning-1024x576.jpg",
      medium:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-01_bottenv%C3%A5ning-300x169.jpg",
      thumbnail:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-01_bottenv%C3%A5ning-150x150.jpg"
    },
    {
      large:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-02_bottenv%C3%A5ning-1024x576.jpg",
      medium:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-02_bottenv%C3%A5ning-300x169.jpg",
      thumbnail:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-02_bottenv%C3%A5ning-150x150.jpg"
    },
    {
      large:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-03_bottenv%C3%A5ning-1024x576.jpg",
      medium:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-03_bottenv%C3%A5ning-300x169.jpg",
      thumbnail:
        "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-03_bottenv%C3%A5ning-150x150.jpg"
    }
  ],
  order: 1,
  postStatus: "publish",
  searchableId: "001",
  title: "Vaktvåningen",
  description:
    "Välkommen in! Trappan upp till vaktvåningen säger något om svårigheten att inta borgen.",
  links: [
    {
      title: "test",
      url: "http://www.google.com/"
    }
  ],
  audio: {
    contentType: "audio",
    created: "2017-11-03T13:47:20.000Z",
    description:
      "\"vakt-svensk\" från 1001 av Kärnan. Släppt: 2014. Spår 1. Genre: Svensk.",
    id: 71197,
    modified: "2017-11-03T13:47:36.000Z",
    title: "Vaktvåningen_svenska",
    url:
      "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/vakt-svensk.mp3"
  },
  beacon: {
    distance: 10,
    id: "",
    nid: "f7826da6bc5b71e0893e",
    position: {
      latitude: 56.0483045,
      longitude: 12.6974978
    }
  }
};

function onSwiperIndexChanged() {}
function onGoToImage() {}
function onGoToLink() {}
function loadAudioFile() {}
function onGoToVideo() {}
function onClosePlayer() {}
function onTogglePlaying() {}
function onSliding() {}
function onSliderValueCompleted() {}

test("Basic GuideObject", () => {
  const tree = renderer
    .create(
      <ObjectView
        contentObject={currentContentObject}
        guideType="guide"
        onSwiperIndexChanged={onSwiperIndexChanged}
        audioButtonDisabled
        videoButtonDisabled
        imageIndex={0}
        onGoToImage={onGoToImage}
        onGoToLink={onGoToLink}
        loadAudioFile={loadAudioFile}
        onGoToVideo={onGoToVideo}
        onClosePlayer={onClosePlayer}
        onTogglePlaying={onTogglePlaying}
        onSliding={onSliding}
        onSliderValueCompleted={onSliderValueCompleted}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
