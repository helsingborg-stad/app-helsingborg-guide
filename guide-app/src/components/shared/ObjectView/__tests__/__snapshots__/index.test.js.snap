// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic GuideObject 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "flex": 1,
      "justifyContent": "flex-end",
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "backgroundColor": "white",
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "aspectRatio": 1.7777777777777777,
            "height": "auto",
            "width": "100%",
          }
        }
      >
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "position": "relative",
              },
              undefined,
            ]
          }
        >
          <RCTScrollView
            activeDotColor="#D35098"
            automaticallyAdjustContentInsets={false}
            autoplay={false}
            autoplayDirection={true}
            autoplayTimeout={2.5}
            bounces={false}
            contentContainerStyle={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                Object {},
              ]
            }
            contentOffset={
              Object {
                "x": 0,
              }
            }
            disableNextButton={false}
            disablePrevButton={false}
            dotColor="white"
            horizontal={true}
            index={0}
            loadMinimal={false}
            loadMinimalSize={1}
            loop={false}
            onIndexChanged={[Function]}
            onMomentumScrollEnd={[Function]}
            onScrollBeginDrag={[Function]}
            onScrollEndDrag={[Function]}
            pagingEnabled={true}
            removeClippedSubviews={true}
            scrollsToTop={false}
            showsButtons={false}
            showsHorizontalScrollIndicator={false}
            showsPagination={true}
            showsVerticalScrollIndicator={false}
          >
            <View>
              <View
                style={
                  Array [
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Image
                      resizeMode="cover"
                      source={
                        Object {
                          "uri": "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-01_bottenv%C3%A5ning-1024x576.jpg",
                        }
                      }
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Image
                      resizeMode="cover"
                      source={
                        Object {
                          "uri": "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-02_bottenv%C3%A5ning-1024x576.jpg",
                        }
                      }
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "height": 1334,
                      "width": 750,
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                <View>
                  <View
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                  >
                    <Image
                      resizeMode="cover"
                      source={
                        Object {
                          "uri": "https://api.helsingborg.se/wp-content/uploads/sites/2/2017/06/1920x1080-03_bottenv%C3%A5ning-1024x576.jpg",
                        }
                      }
                      style={
                        Object {
                          "height": "100%",
                          "width": "100%",
                        }
                      }
                    />
                  </View>
                </View>
              </View>
            </View>
          </RCTScrollView>
          <View
            pointerEvents="none"
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "bottom": 25,
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#D35098",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "borderRadius": 4,
                    "height": 8,
                    "marginBottom": 3,
                    "marginLeft": 3,
                    "marginRight": 3,
                    "marginTop": 3,
                    "width": 8,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "bottom": 25,
              "position": "absolute",
              "right": 25,
              "zIndex": 50,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "stretch",
            "backgroundColor": "white",
          }
        }
      >
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingBottom": 4,
                "paddingHorizontal": 34,
                "paddingTop": 12,
              }
            }
          >
            <Text
              style={
                Object {
                  "fontFamily": "Roboto",
                  "fontSize": 24,
                  "fontWeight": "300",
                  "lineHeight": 30,
                }
              }
            >
              Vaktv√•ningen
            </Text>
          </View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingBottom": 4,
                "paddingHorizontal": 34,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#7f7f7f",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "fontWeight": "400",
                  "lineHeight": 21,
                }
              }
            >
              ID #001
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "minHeight": 60,
            }
          }
        >
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#f2f2f2",
                "borderBottomWidth": 2,
                "borderColor": "#D6D6D6",
                "borderLeftWidth": 1,
                "borderRightWidth": 1,
                "borderTopWidth": 2,
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 46,
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "maxWidth": 120,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  style={
                    Array [
                      Object {
                        "color": "#7B075E",
                        "fontSize": 18,
                      },
                      Array [
                        Object {},
                        Object {
                          "color": "#cecece",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Ôãã
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 0,
                    "paddingLeft": 8,
                  }
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#711A5B",
                        "fontSize": 16,
                        "fontWeight": "500",
                        "lineHeight": 22,
                      },
                      Object {
                        "color": "#cecece",
                      },
                    ]
                  }
                >
                  Play sound
                </Text>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 4,
              "paddingHorizontal": 34,
              "paddingVertical": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "lineHeight": 20,
              }
            }
          >
            V√§lkommen in! Trappan upp till vaktv√•ningen s√§ger n√•got om sv√•righeten att inta borgen.
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "opacity": 1,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#711A5B",
                    "fontSize": 24,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              Ó¢î
            </Text>
            <Text
              style={
                Object {
                  "color": "#711A5B",
                  "fontFamily": "Roboto",
                  "fontSize": 16,
                  "fontWeight": "500",
                  "lineHeight": 23,
                  "marginLeft": 7,
                  "marginRight": 13,
                }
              }
            >
              test
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <Text>
    Audio Player
  </Text>
</View>
`;
